### PR TITLE
RDISCROWD-7573 update error on nonexistent project id in task API

### PR DESCRIPTION
**Pull request by n00rsy**
*Issue number of the reported bug or feature request: [RDISCROWD-7573](https://jira.prod.bloomberg.com/browse/RDISCROWD-7573)*

**Describe your changes**
add project null check after fetching project info in `_preprocess_post_data`

### CODE REVIEW

File: pybossa/api/task.py [hunk 1]
```diff
@@ -102,6 +102,8 @@ def _preprocess_post_data(self, data):
         if 'n_answers' not in data:
             project = project_repo.get(project_id)
             data['n_answers'] = project.get_default_n_answers()
+            if not project:
+                raise BadRequest(f'Non existing project id {project_id}')
```
Comment (kbecker42) [inline pybossa/api/task.py:106]: Are any other details of the request available inside `data[]` that would help identify the caller? If so, those could be included in the exception details.

Comment (n00rsy) [reply pybossa/api/task.py:106]: Please see the ticket comments for planned work on the gigwork listener to add identification for the caller. Thanks!

File: pybossa/api/task.py [hunk 2]
```diff
@@ -102,6 +102,8 @@ def _preprocess_post_data(self, data):
         if 'n_answers' not in data:
             project = project_repo.get(project_id)
             data['n_answers'] = project.get_default_n_answers()
+            if not project:
+                raise NotFound(f'Non existing project id {project_id}')
```
Comment (dchhabda) [inline pybossa/api/task.py:106]: Seems this should be right after line 103

**Review by dchhabda — CHANGES_REQUESTED** [review]
Feel free to add test that will cover newly added condition and exception. Thanks

**Review by dchhabda — APPROVED** [review]

**Pull request merged**